### PR TITLE
Fix MaRaClusterAdapter Exectuable Check Fail in KNIME

### PR DESCRIPTION
**Pull request by Leon-Bichmann**

### CODE REVIEW

File: src/topp/MaRaClusterAdapter.cpp [hunk 1]
```diff
@@ -282,19 +282,6 @@ class MaRaClusterAdapter :
     const String maracluster_executable(getStringOption_("maracluster_executable"));
     writeDebug_(String("Path to the maracluster executable: ") + maracluster_executable, 2);
 
-    if (maracluster_executable.empty())  //TODO? - TOPPBase::findExecutable after registerInputFile_("maracluster_executable"... ???
```
Comment (jpfeuffer) [inline src/topp/MaRaClusterAdapter.cpp]: I think you can leave the check for empty. This should really not happen.

**Comment by Leon-Bichmann** [comment]
so i get the following exit code:

`Executing maracluster ...
Executing: maracluster
FATAL: External invocation of maracluster failed. Standard output and error were:


255`

**Comment by jpfeuffer** [comment]
Hmm unfortunately this is not very clear/satisfying.
Maybe we have to check QProcess::error() to get a better defined error message. Would be best to do it in the runExternalProcess function.

**Comment by Leon-Bichmann** [comment]
I see.. not sure how to do that. Do you have an example code snippet?

**Comment by jpfeuffer** [comment]
Somewhere after qp.waitForFinished() just add:
```
if (qp.error() == QProcess::FailedToStart)
{
  LOG_ERROR << "Process failed to start. Does it exist? Is it executable?" << std::endl;
}
```

**Comment by Leon-Bichmann** [comment]
`Executing maracluster ...
Process failed to start. Does it exist? Is it executable?`

**Comment by jpfeuffer** [comment]
Great. But ;) : could you do it in the runExternalProcess function, so that it is fixed for all the adapters that use this function?
E.g. here:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/APPLICATIONS/TOPPBase.cpp#L1648

**Comment by Leon-Bichmann** [comment]
argh.. ^^ ok now I understand what you mean

**Comment by cbielow** [comment]
how about:
```
LOG_ERROR << "Process '" << executable << "' failed to start. Does it exist? Is it executable?" << std::endl;
return EXTERNAL_PROGRAM_ERROR;
```

**Comment by jpfeuffer** [comment]
I'll merge. Just Sirius and IonizationSim (unrelated).

**Pull request merged**